### PR TITLE
intial setup for the new flow

### PR DESCRIPTION
**Pull request by ivinayakg**
# DRAFT
### What is the change?

Provide a small description in points of what did you change

##### Example :

- Done this
- Made these changes

### Is Development Tested?

- [ ] Yes
- [ ] No

### Before / After Change Screenshots

For visual or interaction changes. Please provide a video / screenshot.

##### Example:

Before :
Image/Video

After :
Image/Video

**Note :**
What to delete and what not to

- Keep the headings and answer the questions.
- Delete the examples
- Delete the description below headings `What is the change?` & `Before/After Screenshots`

### CODE REVIEW

File: api/authCustom/admin.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (pallabez) [inline api/authCustom/admin.py:1]: Do we need this?

File: api/authCustom/models.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django.utils import timezone
+
+
+class Token_Custom(models.Model):
+    """
+    The default authorization token model.
+    """
+    key = models.CharField(_("Key"), max_length=40)
```
Comment (pallabez) [inline api/authCustom/models.py:12]: Wouldn't it be better to call it `auth_key`? Keeping the terms fixed helps throughout the project.

File: api/authCustom/models.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    def save(self, *args, **kwargs):
+        if not self.key:
+            self.key = self.generate_key()
+        return super().save(*args, **kwargs)
+
+    @classmethod
+    def is_invalid(self, createdTime):
+        curr_time = timezone.now().timestamp()
+        exp_time = createdTime + (60*60*24*30)
```
Comment (pallabez) [inline api/authCustom/models.py:24]: Can we move this to a constant file and call it `AUTH_KEY_INVALIDATION_TIME_IN_SECONDS`?

File: api/authCustom/models.py (hunk 2 above)
Comment (pallabez) [reply api/authCustom/models.py:12]: Is it possible to use the `default` field here to generate the `auth_key`?

File: api/authCustom/tests.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+
+class Get_Auth_token_Test(APITestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.factory = APIRequestFactory()
+        super(Get_Auth_token_Test, cls).setUpClass()
+
+    def test_get_auth_token_without_restkey(self):
+        _request = self.factory.post('authToken//get_token')
```
Comment (pallabez) [inline api/authCustom/tests.py:84]: I'm not sure. Shouldn't this have only one `/`?

File: api/authCustom/tests.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        _response = goal_auth_token(_request, userId="vinayak")
+        self.assertEqual(_response.data, 'Wrong REST_KEY')
```
Comment (pallabez) [inline api/authCustom/tests.py:86]: I think we should test the status_code as well.

File: api/authCustom/views.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+@parser_classes([JSONParser])
+def goal_auth_token(request, userId):
+    try:
+        rest_key = request.data.get('rest_key')
+        if rest_key != env('REST_KEY'):
+            raise Exception("Wrong REST_KEY")
+
+        try:
+            token = Token_Custom.objects.get(userId=userId)
+            if token.is_invalid(createdTime=token.created.timestamp()):
```
Comment (pallabez) [inline api/authCustom/views.py:21]: Passing `createdTime` doesn't feel right here. Shouldn't it be already available in token self scope?

File: api/authCustom/views.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                Token_Custom.objects.filter(userId=userId).update(key=Token_Custom.generate_key(), created=timezone.now())
+                token = Token_Custom.objects.get(userId=userId)
+        except Token_Custom.DoesNotExist:
+            token = Token_Custom(userId=userId)
+            token.save()
```
Comment (pallabez) [inline api/authCustom/views.py:26]: Ideally, the view layer shouldn't concern itself with if the token, (or any data) is present or not. It's the responsibility of model layer.

File: api/authCustom/authentications.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+class UserAuthenticationExtender():
+    def __init__(self, key, userId, created):
+        self.key = key 
+        self.userId  = userId
+        self.created = created
+        self.is_authenticated = True
+
+
+class TokenCustomAuthentication(authentication.BaseAuthentication):
```
Comment (pallabez) [inline api/authCustom/authentications.py:14]: Can we call it `AUTH_TOKEN_AUTHENTICATION`?

File: api/authCustom/tests.py (hunk 4 above)
Comment (ivinayakg) [reply api/authCustom/tests.py:84]: so `/g` has some sort of significance in python or django , not 100% sure somewhat similar to how we use `\n or \t` therefore it is there.

File: api/authCustom/views.py (hunk 7 above)
Comment (ivinayakg) [reply api/authCustom/views.py:26]: explain.

**Review by pallabez — APPROVED** [review]
LGTM

**Pull request merged**